Marlin/src/HAL/HAL_STM32F1/timers.h

```diff
@@ -73,7 +73,11 @@ typedef uint16_t hal_timer_t;
 
 #if MB(BIGTREE_SKR_MINI_E3, BIGTREE_SKR_E3_DIP, BTT_SKR_MINI_E3_V1_2, MKS_ROBIN_LITE)
   // SKR Mini E3 boards use PA8 as FAN_PIN, so TIMER 1 is used for Fan PWM.
-  #define SERVO0_TIMER_NUM 3
+  #if defined(STM32_HIGH_DENSITY) || DISABLED(SPEAKER)
+    #define SERVO0_TIMER_NUM 8  // tone.cpp uses Timer 4
+  #else
+    #define SERVO0_TIMER_NUM 3  // tone.cpp uses Timer 8
```

Comment:
We cannot use TIMER 3. See my comment https://github.com/MarlinFirmware/Marlin/pull/15781#issuecomment-549204184 . The TIMER 3 gets disabled by SD card/SPI code and the pointer to its interrupt routine is set to `NULL`. Then a null pointer exception happens when SERVO0 enables the TIMER and an interrupt happens. The printer hangs and can cause a thermal runaway if it hangs during heating.

Reply:
Actually, it would work on my BigTreeTech SKR E3 DIP v1.0 board because it has a hight-density MCU, but if for example `MKS_ROBIN_LITE` had a medium-density MCU and SD card was connected through the same SPI pins, this would cause the described problems when using TIMER 3.

Reply:
Also, as @diabl0w, pointed out, tone.cpp uses:
- Timer 4 on medium-density MCUs and
- Timer 8 on high-density MCUs

So the code (as you changed it) does not even solve the conflict between SERVO0 and speaker. :(

Reply:
@sustmi as you mentioned https://github.com/MarlinFirmware/Marlin/pull/15547#issuecomment-549094349 you didnt want to use timer 4 because it can conflict with stepper motors on medium density, but it is okay for high density. since medium density uses tone in timer 4 anyway, we can use timer 8 for speaker on medium density. then, since we are not worried about timer 4 stepper motors on high density, we can use timer 4 for high density ones

Reply:
@sustmi sorry to keep pinging you on this, but do you change SERVO0_TIMER_NUM to 4 when compiling Marlin since this was merged backwards? @thinkyhead Timer 8 is occupied by tone.cpp on high density mcu's, so we should be using timer 4, I believe (opposite of what is merged in the committ)